NitroxServer/Serialization/World/WorldPersistence.cs

```diff
@@ -28,16 +28,23 @@ public class WorldPersistence
         private string FileEnding => Serializer?.FileEnding ?? "";
 
         private readonly ServerProtoBufSerializer protoBufSerializer;
+        private readonly ServerJsonSerializer jsonSerializer;
         private readonly RandomStartGenerator randomStart;
         private readonly ServerConfig config;
 
         public WorldPersistence(ServerProtoBufSerializer protoBufSerializer, ServerJsonSerializer jsonSerializer, ServerConfig config, RandomStartGenerator randomStart)
         {
             this.protoBufSerializer = protoBufSerializer;
+            this.jsonSerializer = jsonSerializer;
             this.randomStart = randomStart;
             this.config = config;
 
-            Serializer = config.SerializerMode == ServerSerializerMode.PROTOBUF ? protoBufSerializer : jsonSerializer;
+            UpdateSerializer(config.SerializerMode);
+        }
+
+        internal void UpdateSerializer(ServerSerializerMode mode)
+        {
+            Serializer = (mode == ServerSerializerMode.PROTOBUF) ? protoBufSerializer : jsonSerializer;
         }
```

Comment:
This should error out `WorldPersistenceTest.Setup()`. Are your sure all tests are still working?